Dockerfile: golang image version bump from 1.13 to 1.14.1

etungsten opened the pull request:
<!--
Tips:
- Please read the CONTRIBUTING document to understand our process and our requests for PRs.
- Please file an issue before creating a PR so we can discuss the change and confirm it's not already being worked on.
-->

**Issue number:**
N/A


**Description of changes:**
Bumps the golang image version from 1.13 to 1.14.1 to match the
bottlerocket's SDK golang toolchain version


**Testing done:**
`make container` builds the update operator container image successfully.


**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

jahkeup commented:
@etungsten not sure why this is showing a diff - we got this in earlier when we went to merge in #21. I missed that the lingering `golang:$version` image didn't get bumped to match the sdk's go toolchain version - can you tackle bumping that?

etungsten committed:
Dockerfile: golang version bump from 1.13 to 1.14.1

Bumps the golang image version from 1.13 to 1.14.1 to make the
bottlerocket's SDK golang toolchain version

etungsten commented:
Updated the PR to bump the golang image version.

jahkeup reviewed (approved):
LGTM. :ship: :shipit:
